packages/next/src/client/components/http-access-fallback/error-boundary.tsx

```diff
@@ -112,25 +110,42 @@ class HTTPAccessFallbackErrorBoundary extends React.Component<
   }
 
   render() {
-    const { notFound } = this.props
-    if (this.state.triggeredStatus === HTTPErrorStatus.NOT_FOUND) {
+    const { notFound, forbidden, unauthorized, children } = this.props
+    const { triggeredStatus } = this.state
+    if (triggeredStatus) {
+      const isNotFound =
+        triggeredStatus === HTTPAccessErrorStatus.NOT_FOUND && notFound
+      const isForbidden =
+        triggeredStatus === HTTPAccessErrorStatus.FORBIDDEN && forbidden
+      const isUnauthorized =
+        triggeredStatus === HTTPAccessErrorStatus.UNAUTHORIZED && unauthorized
+
+      // If there's no matched boundary in this layer, keep throwing the error by rendering the children
+      if (!(isNotFound || isForbidden || isUnauthorized)) {
+        return children
+      }
+
       return (
         <>
           <meta name="robots" content="noindex" />
           {process.env.NODE_ENV === 'development' && (
             <meta name="next-error" content="not-found" />
           )}
-          {notFound}
+          {isNotFound ? notFound : null}
+          {isForbidden ? forbidden : null}
+          {isUnauthorized ? unauthorized : null}
```

Comment:
seems like this should be represented differently as it's never possible for 1+ of these to be active at a time right? 

maybe something like:

```
const errorComponents = {
  [HTTPAccessErrorStatus.NOT_FOUND]: notFound,
  [HTTPAccessErrorStatus.FORBIDDEN]: forbidden,
  [HTTPAccessErrorStatus.UNAUTHORIZED]: unauthorized,
};

const errorComponent = errorComponents[triggeredStatus] ?? null;
```